server/constants/vars.js

```diff
@@ -62,3 +64,49 @@ export const ALL_SUBJECTS_MONGO_PROJECTION = {
   _id: 0,
   subject: 1,
 }
+
+export const DEFAULT_CONFIG_AT_SIGNUP = (uid) => ({
+  owner: uid,
+  config: defaultUserConfig,
+  name: 'default',
+  type: 'matrix',
+  readers: [uid],
+  created: new Date().toUTCString(),
+})
+
+export const SIGN_IN_MONGO_PROJECTION = {
+  _id: 0,
+  uid: 1,
+  display_name: 1,
+  acl: 1,
+  role: 1,
+  icon: 1,
+  mail: 1,
+  access: 1,
+  account_expires: 1,
+}
+
+export const INITIAL_ROLE = 'member'
+
+export const SIGN_UP_UPDATE_DATA = (overrides) => ({
+  display_name: '',
+  title: '',
+  department: '',
+  company: '',
+  mail: '',
+  member_of: '',
+  bad_pwd_count: 0,
+  lockout_time: 0,
+  last_logoff: Date.now(),
+  last_logon: Date.now(),
+  account_expires: null,
+  password: '',
+  ldap: false,
+  force_reset_pw: false,
+  realms: '',
+  icon: '',
+  access: [],
+  blocked: false,
+  role: INITIAL_ROLE,
+  ...overrides,
+})
```

Comment:
should this go in a UserModel?

```suggestion
const UserModel = {
  default: (overrides = {}) => ({
    display_name: '',
    title: '',
    department: '',
    company: '',
    mail: '',
    member_of: '',
    bad_pwd_count: 0,
    lockout_time: 0,
    last_logoff: Date.now(),
    last_logon: Date.now(),
    account_expires: null,
    password: '',
    ldap: false,
    force_reset_pw: false,
    realms: '',
    icon: '',
    access: [],
    blocked: false,
    role: INITIAL_ROLE,
    ...overrides,
  }),
}

export default UserModel
```

Reply:
[de48040](https://github.com/AMP-SCZ/dpdash/pull/293/commits/de48040cb5b39f723d4ad1877be815b064027992)

Reply:
this is making more sense now thank you